src/eval.ml

```diff
@@ -187,34 +196,55 @@ and tree_walk : dtree -> stack -> (term * stack) option = fun tree stk ->
                 env.(slot) <- TE_Some(b)
               else
                 let b = lift vars.(pos) in
-                let xs = Array.map (fun e -> VarMap.find e fresh_vars) xs in
+                let xs = Array.map (fun e -> IntMap.find e id_vars) xs in
                 env.(slot) <- TE_Some(Bindlib.unbox (Bindlib.bind_mvar xs b))
         in
         List.iter fn env_builder;
         (* Actually perform the action. *)
         Some(Bindlib.msubst act env, stk)
-    | Cond({ ok ; cond ; fail })                          ->
+    | Cond({ok; cond; fail})                              ->
         let next =
           match cond with
           | CondNL(i, j) ->
               if eq_modulo vars.(i) vars.(j) then ok else fail
           | CondFV(xs,i) ->
-              let xs = Array.map (fun e -> VarMap.find e fresh_vars) xs in
+              let allowed =
+                (* Term variables allowed in the term. *)
+                let _, dummy = IntMap.choose id_vars in
+                Array.make (Array.length xs) dummy in
+              let lightened_map =
```

Comment:
Why building a map since the values of the domain are ignored in the function [fn] used in [only_allowed]?

Reply:
Which domain? [only_allowed] doesn't ignore anything on its own.